### DISCUSSION TITLE
Single Fetch - missing action data on error

### DISCUSSION THREAD

**Discussion by ziimakc** [remix-run/remix#9342] [Q&A] [answered]
I return from action following data:
```
args.response!.status = 400;
return { data: null, err: {message: "smth"}};
```

But I don't get an error in component:
```
const data = useActionData<typeof action>();
console.error(123); // logged
console.error(data);  // undefined
```

I'm a bit confused about it, what is the problem? How do I get error information to show error message?

In network tab I can confirm only post request is made and response data of it includes an error I returned (in special format)

**Comment by sergiodxa** [comment]
To submit the request to that action, are you using Form/useSubmit or fetcher.Form/fetcher.submit?

**Reply by ziimakc** [reply]
```
<fetcher.Form method="post">
						<Button
							onClick={(event) => {
								fetcher.submit(event.currentTarget.form, {
									method: "POST",
								});
								inEditSet(false);
							}}
						>
							Save
						</Button>
				</fetcher.Form>
```

Previously it worked fine

**Reply by sergiodxa** [reply, marked as answer]
If you're using a fetcher, then the result of the action will be accesible in `fetcher.data`, not using useActionData. This is the same with and without Single Fetch.

**Reply by ziimakc** [reply]
hm I'm pretty sure it was accessible in `useActionData` before single fetch, maybe I'm wrong, thanks
